services/fileParser.effect.ts

```
@@ -0,0 +1,657 @@
+      if (existingLine) {
+        const gap = x - existingLine.lastX;
+
+        // Smart spacing: gap > 4px = add space
+        const newText =
+          gap > 4
+            ? existingLine.text + " " + text
+            : existingLine.text + text;
+
+        // Create new line (immutable)
+        const index = lines.indexOf(existingLine);
+        lines = [
```

Biome (2.1.2)

[error] Can't assign lines because it's a constant.

This is where the variable is defined as constant.

Unsafe fix: Replace const with let if you assign it to a new value.


(lint/correctness/noConstAssign)